apps/prairielearn/src/pages/authCallbackLti/authCallbackLti.js

```diff
@@ -105,23 +108,32 @@ router.post(
     }
     const authName = parameters.lis_person_name_full || fallbackName;
 
-    const userResult = await sqldb.callAsync('users_select_or_insert_and_enroll_lti', [
-      authUid,
-      authName,
-      ltiresult.course_instance_id,
-      parameters.user_id,
-      parameters.context_id,
-      res.locals.req_date,
-    ]);
-    if (!userResult.rows[0].has_access) {
+    const userResult = await sqldb.callOptionalRow(
+      'users_select_or_insert_and_enroll_lti',
+      [
+        authUid,
+        authName,
+        ltiResult.course_instance_id,
+        parameters.user_id,
+        parameters.context_id,
+        res.locals.req_date,
+      ],
+      z.object({
+        user_id: IdSchema,
+        has_access: z.boolean(),
+      }),
+    );
+    if (!userResult?.has_access) {
       throw new HttpStatusError(403, 'Access denied');
     }
 
-    const tokenData = {
-      user_id: result.rows[0].user_id,
-      authn_provider_name: 'LTI',
-    };
-    const pl_authn = generateSignedToken(tokenData, config.secretKey);
+    const pl_authn = generateSignedToken(
+      {
+        user_id: userResult.user_id,
+        authn_provider_name: 'LTI',
+      },
+      config.secretKey,
+    );
```

Comment:
A fun bit of trivia: the only reason this didn't break logins is that we're correctly reading the user information from the session!